File: utbot-framework/src/main/java/org/utbot/quickcheck/internal/generator/ZilchGenerator.java


package org.utbot.quickcheck.internal.generator;

import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator;
import org.utbot.framework.plugin.api.UtModel;
import org.utbot.quickcheck.generator.GenerationStatus;
import org.utbot.quickcheck.generator.Generator;
import org.utbot.quickcheck.internal.Zilch;
import org.utbot.quickcheck.random.SourceOfRandomness;

import static org.utbot.external.api.UtModelFactoryKt.classIdForType;

public class ZilchGenerator extends Generator<Zilch> {
    public ZilchGenerator() {
        super(Zilch.class);
    }

    @Override public UtModel generate(
        SourceOfRandomness random,
        GenerationStatus status) {

        return UtModelGenerator.getUtModelConstructor().construct(Zilch.INSTANCE, classIdForType(Zilch.class));
    }
}

File: utbot-framework/src/main/kotlin/org/utbot/quickcheck/internal/generator/ZilchGenerator.kt
package org.utbot.quickcheck.internal.generator

import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator.utModelConstructor
import org.utbot.external.api.classIdForType
import org.utbot.framework.plugin.api.UtModel
import org.utbot.framework.plugin.api.UtNullModel
import org.utbot.framework.plugin.api.util.objectClassId
import org.utbot.quickcheck.generator.GenerationStatus
import org.utbot.quickcheck.generator.Generator
import org.utbot.quickcheck.internal.Zilch
import org.utbot.quickcheck.random.SourceOfRandomness

class ZilchGenerator : Generator(Zilch::class.java) {
    override fun generate(
        random: SourceOfRandomness,
        status: GenerationStatus
    ): UtModel {
        return UtNullModel(objectClassId)//utModelConstructor.construct(Zilch, classIdForType(Zilch::class.java))
    }
}